src/Core/Managed/Shared/Extensibility/Metrics/MetricExtractorTelemetryProcessorBase.cs

```diff
@@ -0,0 +1,164 @@
+        private string _extractorName = null;
+
+        public MetricExtractorTelemetryProcessorBase(ITelemetryProcessor nextProcessorInPipeline)
+            : this(nextProcessorInPipeline, metricTelemetryMarkerKey: null, metricTelemetryMarkerValue: null)
+        {
+        }
+
+        public MetricExtractorTelemetryProcessorBase(ITelemetryProcessor nextProcessorInPipeline, string metricTelemetryMarkerKey, string metricTelemetryMarkerValue)
+        {
+            _nextProcessorInPipeline = nextProcessorInPipeline;
+
+            if (metricTelemetryMarkerKey != null)
+            {
+                metricTelemetryMarkerKey = metricTelemetryMarkerKey.Trim();
+                if (metricTelemetryMarkerKey.Length == 0)
+                {
+                    metricTelemetryMarkerKey = null;
+                }
+            }
+
+            if (metricTelemetryMarkerValue != null)
+            {
+                metricTelemetryMarkerValue = metricTelemetryMarkerValue.Trim();
+            }
+            else
+            {
+                metricTelemetryMarkerValue = String.Empty;
+            }
+
+            MetricTelemetryMarkerKey = metricTelemetryMarkerKey;
+            MetricTelemetryMarkerValue = metricTelemetryMarkerValue;
+        }
+
+        public string MetricTelemetryMarkerKey { get; private set; }
+
+        public string MetricTelemetryMarkerValue { get; private set; }
+
+        protected MetricManager MetricManager
+        {
+            get { return _metricManager; }
+        }
+
+        protected virtual string ExtractorName
+        {
+            get
+            {
+                string name = _extractorName;
+                if (name == null)
+                {
+                    name = this.GetType().FullName;
+                    _extractorName = name;  // benign race
+                }
+                return name;
+            }
+        }
+
+        void ITelemetryModule.Initialize(TelemetryConfiguration configuration)
+        {
+            TelemetryClient telemetryClient = (configuration == null)
+                                                    ? new TelemetryClient()
+                                                    : new TelemetryClient(configuration);
+            if (MetricTelemetryMarkerKey != null)
+            {
+                telemetryClient.Context.Properties[MetricTelemetryMarkerKey] = MetricTelemetryMarkerValue;
+            }
+            
+            _metricManager = new MetricManager(telemetryClient);
+
+            Initialize(configuration);
+        }
+
+        void ITelemetryProcessor.Process(ITelemetry item)
+        {
+            if (item == null)
+            {
+                InvokeNextProcessor(item);
+                return;
+            }
+
+            try
+            {
+                Process(item);
+                AddExtractorInfo(item);
+            }
+            catch(Exception ex)
+            {
+                // Is this te right way to log erroes here?
```

Comment:
nitpick: "te" = "the"
The answer is that you should usually make a specific message in the CoreEventSource rather than calling Log.LogError.